api/pkg/block/service.go

```diff
@@ -74,24 +235,95 @@ func (s *APIService) CreateVolume(request *restful.Request, response *restful.Re
 	response.WriteEntity(res.Volume)
 }
 
-// List all volumes
-func (s *APIService) ListVolumes(request *restful.Request, response *restful.Response) {
-	if !policy.Authorize(request, response, "volume:list") {
+func (s *APIService) UpdateVolume(request *restful.Request, response *restful.Response) {
+	if !policy.Authorize(request, response, "volume:put") {
+		return
+	}
+	log.Info("Received request for updating volume.")
+
+	volume := &model.Volume{}
+
+	err := request.ReadEntity(&volume)
+	if err != nil {
+		log.Errorf("Failed to read request body err: \n", err)
+		response.WriteError(http.StatusInternalServerError, err)
 		return
 	}
-	log.Infof("Received request to list volumes")
 
-	backend := request.QueryParameter(common.REQUEST_PATH_BACKEND_ID)
+	vol := &block.Volume{
+		Description: volume.Description,
+	}
+
+	if volume.Metadata != nil || len(volume.Metadata) != 0 {
+		metadata, err := utils.ConvertMapToStruct(volume.Metadata)
+		if err != nil {
+			log.Errorf("Failed to parse metadata: %v\n", volume.Metadata, err)
```

Comment:
It will be a good idea to move the Error messages to a common file , later it can be use for internationaliation .

Reply:
Yes, There is lot scope for making generalization for logging, exceptions, Error handling for multi-cloud.
Thanks for your valuable suggestion.